### PR TITLE
XDR iocs integration- 'XSOAR Comment Field Exporting To XDR' parameter is handled incorrectly

### PR DESCRIPTION
**Pull request by rshunim**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11945

## Description
Due to a change in the parameter type (from type 0 - text, to type 16 - list), we encounter an error when updates are sent from XDR to XSOAR.
This error occurs because a list is being used as a key in a dictionary.
Handling it in this fix.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by dorschw — CHANGES_REQUESTED** [review]
DM

**Comment by rshunim** [comment]
The fix was checked with @tcarmeli1 and seems to resolve the issue.

**Comment by rshunim** [comment]
I'll fix the UTs according to the fix and wait with merging the PR since there is an issue with mirroring the 'comments' from XDR to XSOAR

@dorschw FYI

**Comment by rshunim** [comment]
After digging with @tcarmeli1 in the logs,
Continuing to investigate two things:
One, why aren't the comments mirroring to XSOAR from XDR?
Two, why is there a link as a comment even when not configured?

**Comment by rshunim** [comment]
@tcarmeli1 @dorschw please review it both of you

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py [hunk 1]
```diff
@@ -294,38 +295,52 @@ def demisto_types_to_xdr(_type: str) -> str:
...
+    # a regular comment for the given field name, "comments" as a default
+    single_comment_for_field_name = parse_demisto_single_comments(ioc, comment_field_name, comments_as_tags)
+    if single_comment_for_field_name:
+        comments.extend(single_comment_for_field_name)
+
+    # if the flag is True, add a link as a comment
+    if Client.add_link_as_a_comment:
+        comments.extend(create_an_indicator_link(ioc))
+
+    return (comments or ['']) if comments_as_tags else [', '.join(comments)]
```
Comment (tcarmeli1) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:335]: What happens when there is an indicator link? it is being attached as a tag?

**Review by tcarmeli1 — CHANGES_REQUESTED** [review]
Great work! :)

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py [hunk 2]
```diff
@@ -294,38 +295,52 @@ def demisto_types_to_xdr(_type: str) -> str:
...
-            parsing = parse_demisto_single_comments(ioc, field, comments_as_tags)
-            if parsing:
-                comments.extend(parsing)
-        return [', '.join(comments)]
 
-    # else return single field
-    return parse_demisto_single_comments(ioc, comment_field_name, comments_as_tags)
+    comments = []
+    # a regular comment for the given field name, "comments" as a default
+    single_comment_for_field_name = parse_demisto_single_comments(ioc, comment_field_name, comments_as_tags)
```
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:327]:
```suggestion
    comment = parse_demisto_single_comments(ioc, comment_field_name, comments_as_tags)
```

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py (hunk 1 above)
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:335]: let's change it to an if/else for readability

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py [hunk 3]
```diff
@@ -853,8 +887,11 @@ def main():  # pragma: no cover
         Client.tag = tag
     if xsoar_severity_field := params.get('xsoar_severity_field'):
         Client.xsoar_severity_field = to_cli_name(xsoar_severity_field)
-    if xsoar_comment_field := params.get('xsoar_comments_field'):
-        Client.xsoar_comments_field = xsoar_comment_field
+
+    # in case of xsoar_comment_field is an empty list -> the Client.xsoar_comments_field is defined to "comments" by default
+    if xsoar_comment_field := argToList(params.get('xsoar_comments_field')):
+        xsoar_comment_field, add_link_as_a_comment = parse_xsoar_field_name_and_link(xsoar_comment_field)
+        Client.xsoar_comments_field, Client.add_link_as_a_comment = xsoar_comment_field, add_link_as_a_comment
```
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py:894]:
```suggestion
        Client.xsoar_comments_field, Client.add_link_as_a_comment = parse_xsoar_field_name_and_link(xsoar_comment_field)
```

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs_test.py [hunk 4]
```diff
@@ -513,7 +528,8 @@ def test_xdr_expiration_to_demisto(self, xdr_expiration, demisto_expiration):
                 'RULE_SEVERITY': 'SEV_010_INFO', 'NUMBER_OF_HITS': 0, 'RULE_SOURCE': 'XSOAR TIM', 'RULE_COMMENT': '',
                 'RULE_STATUS': 'DISABLED', 'BS_STATUS': 'DONE', 'BS_TS': 1591165801230, 'BS_RETRIES': 1,
                 'RULE_EXPIRATION_TIME': -1, 'IOC_TYPE': 'HASH',
-                'RULE_INDICATOR': 'fa66f1e0e318b6d7b595b6cee580dc0d8e4ac38fbc8dbfcac6ad66dbe282832e', 'REPUTATION': 'GOOD',    # noqa: E501
+                'RULE_INDICATOR': 'fa66f1e0e318b6d7b595b6cee580dc0d8e4ac38fbc8dbfcac6ad66dbe282832e', 'REPUTATION': 'GOOD',
```
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs_test.py:531]: undo

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs_test.py [hunk 5]
```diff
@@ -534,7 +550,8 @@ def test_xdr_expiration_to_demisto(self, xdr_expiration, demisto_expiration):
                 'RULE_ID': 861, 'RULE_INSERT_TIME': 1591165763753, 'RULE_MODIFY_TIME': 1591166095668,
                 'RULE_SEVERITY': 'SEV_010_INFO', 'NUMBER_OF_HITS': 0, 'RULE_SOURCE': 'XSOAR TIM', 'RULE_COMMENT': '',
                 'RULE_STATUS': 'DISABLED', 'BS_STATUS': 'DONE', 'BS_TS': 1591165801784, 'BS_RETRIES': 1,
-                'RULE_EXPIRATION_TIME': -1, 'IOC_TYPE': 'DOMAIN_NAME', 'RULE_INDICATOR': 'test.com', 'REPUTATION': 'GOOD',    # noqa: E501
+                'RULE_EXPIRATION_TIME': -1, 'IOC_TYPE': 'DOMAIN_NAME', 'RULE_INDICATOR': 'test.com', 'REPUTATION': 'GOOD',
+                # noqa: E501
```
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs_test.py:554]: undo

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs_test.py [hunk 6]
```diff
@@ -1268,3 +1288,61 @@ def test_get_indicators(mock_IndicatorsSearcher,
...
+        -  invalid xsoar_comment_field and the expected_result
+    When:
+        -  parsing xsoar_comment_field by our logic
+    Then:
+        - verify the function throws a DemistoException with informative message
+    """
+    import XDR_iocs
+    try:
+        XDR_iocs.parse_xsoar_field_name_and_link(xsoar_comment_field)
+    except DemistoException as e:
```
Comment (dorschw) [inline Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs_test.py:1347]: try `with pytest.raises(DemistException, match=informative_message)`

File: Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py (hunk 1 above)
Comment (rshunim) [reply Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py]: it's just a comment in XDR

**Review by dorschw — APPROVED** [review]

**Pull request merged**